pallets/settlement/src/lib.rs

```diff
@@ -1098,6 +1098,21 @@ impl<T: Trait> Module<T> {
         let weight_for_execution = if Self::instruction_affirms_pending(instruction_id) > 0 {
             // Instruction rejected. Unlock any locked tokens and mark receipts as unused.
             // NB: Leg status is not updated because Instruction related details are deleted after settlement in any case.
+            for (leg_id, _) in legs.iter() {
```

Comment:
We also need to mark receipts as unclaimed when the instruction is approved but fails settlement. (compliance failure)

Reply:
Added